##杂交牛项目

**mapping（新脚本）
```
/storage/public/home/2024050455/07.yunling_Angus_Wagyu/01.mapping/mapping.py
```

**genotype
```
/storage/public/home/2024050455/07.yunling_Angus_Wagyu/05.filter/produceShell_new.py

/storage/public/home/2024050455/07.yunling_Angus_Wagyu/05.filter/cat.sh
```

提杂合子位点
```
bcftools query -f '%CHROM\t%POS\t%REF\t%ALT[\t%GT]\n' YH.SNP.vcf.gz > YH.SNP.GT.file.txt
awk -F'\t' '{
    if ($6 == "0/1" || $6 == "1/0") {
        print $0
    }
}' YH.SNP.GT.file.txt > YH_het.txt

#杂合位点符合亲子关系的提出来

awk -F'\t' '{
    mother = $5;
    father = $7;
    if ((mother == "0/0" && father == "1/1") ||
        (mother == "1/1" && father == "0/0") ||
        (mother == "0/1" && father == "1/1") ||
	(mother == "1/1" && father == "0/1") ||
	(mother == "0/0" && father == "0/1") ||
	(mother == "0/1" && father == "0/0")) {
        print $0
    }
}' YH_het.txt > YH_filtered.txt

#创建bed用来mask基因组

cat YH_filtered.txt | awk '{print $1"\t"$2-1"\t"$2}' > YH_SNP.het.bed

```

**创建split的文件
```
#!/bin/bash

# 定义样本索引
MOTHER=0    # DNA_B256_1
CHILD=1     # DNA_B672_1  
FATHER=2    # yunling-fathersemen.1

# 检查样本顺序
echo "检查样本顺序："
bcftools query -l YH.SNP.vcf.gz
echo ""

# 创建输出文件
> snpsplit_final.txt
echo -e "SNP-ID\tChromosome\tPosition\tStrand\tRef/SNP" > header.txt

# 模式1: 母亲0/0 父亲1/1 孩子0/1
echo "筛选模式1: 母亲0/0 父亲1/1 孩子0/1"
bcftools view -i "GT[$MOTHER]='0/0' && GT[$FATHER]='1/1' && GT[$CHILD]='0/1'" \
  YH.SNP.vcf.gz | \
bcftools query -f '%CHROM\_%POS\t%CHROM\t%POS\t1\t%REF/%ALT\n' > pattern1.txt
echo "找到 $(wc -l < pattern1.txt) 个SNP"

# 模式2: 母亲1/1 父亲0/0 孩子0/1
echo "筛选模式2: 母亲1/1 父亲0/0 孩子0/1"
bcftools view -i "GT[$MOTHER]='1/1' && GT[$FATHER]='0/0' && GT[$CHILD]='0/1'" \
  YH.SNP.vcf.gz | \
bcftools query -f '%CHROM\_%POS\t%CHROM\t%POS\t1\t%ALT/%REF\n' > pattern2.txt
echo "找到 $(wc -l < pattern2.txt) 个SNP"

# 模式3: 母亲0/1 父亲1/1 孩子0/1
echo "筛选模式3: 母亲0/1 父亲1/1 孩子0/1"
bcftools view -i "GT[$MOTHER]='0/1' && GT[$FATHER]='1/1' && GT[$CHILD]='0/1'" \
  YH.SNP.vcf.gz | \
bcftools query -f '%CHROM\_%POS\t%CHROM\t%POS\t1\t%REF/%ALT\n' > pattern3.txt
echo "找到 $(wc -l < pattern3.txt) 个SNP"

# 模式4: 母亲1/1 父亲0/1 孩子0/1
echo "筛选模式4: 母亲1/1 父亲0/1 孩子0/1"
bcftools view -i "GT[$MOTHER]='1/1' && GT[$FATHER]='0/1' && GT[$CHILD]='0/1'" \
  YH.SNP.vcf.gz | \
bcftools query -f '%CHROM\_%POS\t%CHROM\t%POS\t1\t%ALT/%REF\n' > pattern4.txt
echo "找到 $(wc -l < pattern4.txt) 个SNP"

# 模式5: 母亲0/0 父亲0/1 孩子0/1
echo "筛选模式5: 母亲0/0 父亲0/1 孩子0/1"
bcftools view -i "GT[$MOTHER]='0/0' && GT[$FATHER]='0/1' && GT[$CHILD]='0/1'" \
  YH.SNP.vcf.gz | \
bcftools query -f '%CHROM\_%POS\t%CHROM\t%POS\t1\t%REF/%ALT\n' > pattern5.txt
echo "找到 $(wc -l < pattern5.txt) 个SNP"

# 模式6: 母亲0/1 父亲0/0 孩子0/1
echo "筛选模式6: 母亲0/1 父亲0/0 孩子0/1"
bcftools view -i "GT[$MOTHER]='0/1' && GT[$FATHER]='0/0' && GT[$CHILD]='0/1'" \
  YH.SNP.vcf.gz | \
bcftools query -f '%CHROM\_%POS\t%CHROM\t%POS\t1\t%ALT/%REF\n' > pattern6.txt
echo "找到 $(wc -l < pattern6.txt) 个SNP"

# 合并所有模式
echo ""
echo "合并所有模式..."
cat pattern1.txt pattern2.txt pattern3.txt \
    pattern4.txt pattern5.txt pattern6.txt | sort -u > all_snps.txt

# 添加表头
cat header.txt all_snps.txt > snpsplit_final.txt

# 统计信息
total_snps=$(wc -l < all_snps.txt)
echo "========================================"
echo "完成！总共找到 $total_snps 个SNP"
echo "输出文件: snpsplit_final.txt"
echo ""
echo "各模式统计："
echo "模式1: $(wc -l < pattern1.txt)"
echo "模式2: $(wc -l < pattern2.txt)" 
echo "模式3: $(wc -l < pattern3.txt)"
echo "模式4: $(wc -l < pattern4.txt)"
echo "模式5: $(wc -l < pattern5.txt)"
echo "模式6: $(wc -l < pattern6.txt)"
echo "========================================"

# 显示前几行结果
echo ""
echo "前5行结果："
head -5 snpsplit_final.txt
```

**mask基因组

```
bedtools maskfasta -fi /storage/public/home/2022060229/reference/01.ARS-UCD1.2_Btau5.0.1Y/ARS-UCD1.2_Btau5.0.1Y.fa \
       -bed 
       
#输入文件是bed        /storage/public/home/2024050455/07.yunling_Angus_Wagyu/05.filter/YH_SNP.het.bed -mc N \
       -fo YH.masked.fa
jsub -R "span[hosts=1]" -q normal -J bowtie2_index -n 2 -o ./log/bowtie2.o -e ./log/bowtie2.e "bowtie2-build YH.masked.fa YAW.masked.fa"
jsub -R "span[hosts=1]" -q normal -J picard_index -n 2 -o ./log/picard.o -e ./log/picard.e "java -jar /storage/public/home/2022060229/software/jar-pkg/picard.jar CreateSequenceDictionary REFERENCE=YH.masked.fa OUTPUT=YH.masked.dict"
jsub -R "span[hosts=1]" -q normal -J bwa_index -n 2 -o ./log/bwa.o -e ./log/bwa.e "bwa index -a bwtsw YH.masked.fa"
jsub -R "span[hosts=1]" -q normal -J samtools_index -n 2 -o ./log/samtools.o -e ./log/samtools.e "samtools faidx YH.masked.fa"
jsub -R "span[hosts=1]" -q normal -J star_index -n 2 -o ./log/star.o -e ./log/star.e "/storage/public/home/2022060229/software/STAR_2.7.11b/Linux_x86_64_static/STAR --runThreadN 10 --runMode genomeGenerate --genomeDir STAR_index --genomeFastaFiles YH.masked.fa --sjdbGTFfile /storage/public/home/2022060229/reference/05.Sscrofa11.1/Sus_scrofa.Sscrofa11.1.110.chr.gtf"
```

###测序子代数据8个lncRNA 做以下处理
